deltachat-ffi/src/lib.rs

```diff
@@ -949,13 +973,12 @@ pub unsafe extern "C" fn dc_get_chat_media(
         from_prim(or_msg_type3).expect(&format!("incorrect or_msg_type3 = {}", or_msg_type3));
     ffi_context
         .with_inner(|ctx| {
-            let arr = dc_array_t::from(chat::get_chat_media(
-                ctx,
-                chat_id,
-                msg_type,
-                or_msg_type2,
-                or_msg_type3,
-            ));
+            let arr = dc_array_t::from(
+                chat::get_chat_media(ctx, chat_id, msg_type, or_msg_type2, or_msg_type3)
+                    .iter()
+                    .map(|msg_id| msg_id.to_u32())
+                    .collect::<Vec<u32>>(),
```

Comment:
out of curiosity: is there a way to tell the type system to do the u32 conversion automatically if we collect a Vec<u32>?

Reply:
AFAIK it does do this automatically, I could have written `.collect::<Vec<_>>()` I think because of the input to already has the type specified.  But I liked having this explicit.

Oh way, maybe you meant to omit the `.map(...)` line.  Hmm, I don't know.  My guess would be not.